### PR TITLE
feat: use file based OLM catalog

### PR DESCRIPTION
**Pull request by raelga**
This pull request includes significant changes to the GitHub Actions workflow and the Makefile for the `prometheus-exporter-operator` project. The changes primarily focus on improving the release process, containerization, and metadata management.

### GitHub Actions Workflow Improvements:
* Updated the trigger for the release workflow to run on tags matching the pattern `v*` instead of branches. (`.github/workflows/release.yaml`)
* Added steps to set up Docker metadata, QEMU, and Docker Buildx, and updated the build and push steps to use these new configurations, adding support for ARM64 and AMD64 architectures. (`.github/workflows/release.yaml`)

### Makefile Enhancements:
* Introduced environment variable handling for versioning when running in GitHub Actions. (`Makefile`)
* Replaced `docker-build` and `docker-push` targets with `container-build` and `container-push` to support multiple architectures using Docker Buildx. (`Makefile`)
* Added new targets for managing catalog entries and validating the catalog. (`Makefile`)
* Updated the `bundle-build` and `bundle-push` targets to use the new container build process. (`Makefile`)
* Improved bundle catalog manifest generation to inherit alpha and stable releases from $VERSION. `Makefile`)

### Metadata and Dockerfile Updates:
* Updated bundle and catalog Dockerfiles to align with the new build process and metadata requirements. (`bundle.Dockerfile`, `catalog/Dockerfile`) [[1]](diffhunk://#diff-5fc35123fcf790367bb280d63e74ddb015a7393edde91df0ebb77eb7e9f90f18L8-R8) [[2]](diffhunk://#diff-299301ced7df38f1c53529b081fa0b17d73f18968fe67871188f8e16dce3af1dR1-R14)
* Adjusted metadata annotations to reflect the new versioning and channel configurations. (`bundle/metadata/annotations.yaml`)

## File based OLM catalog

* Updates the catalog workflow with [the latest iteration of OLM’s catalog format](https://olm.operatorframework.io/docs/reference/file-based-catalogs/)

These changes collectively enhance the automation, compatibility, consistency, and reliability of the release process for the `prometheus-exporter-operator` project.

## Notes

/kind feature
/priority important-longterm
/assign

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -2,8 +2,8 @@ name: release
 
 on:
   push:
-    branches:
-      - main
+    tags:
+      - "v*"
```
Comment (slopezz) [inline .github/workflows/release.yaml:6]: why this change?

Right now it was pretty easy as documented at https://github.com/3scale-ops/prometheus-exporter-operator/blob/main/docs/release.md#stable, you dont need to add any manual tag 
..., you change the bundle files... and if the GH action detect there is a change on the version, execute the release process

Comment (raelga) [reply .github/workflows/release.yaml]: Release file has been removed, will be implemented in the next PR as discussed internally.

**Review by slopezz — APPROVED** [review]

**Review by roivaz — APPROVED** [review]
